### PR TITLE
refactor(llm): move LangChain implementations into integrations/langchain/

### PR DESCRIPTION
**Pull request by Pouyanpi**
Part of the LangChain decoupling stack:
1. stack-1: canonical types (#1745)
2. stack-2: adapter and framework registry (#1759)
3. stack-3: pipeline rewrite + caller migration (#1760)
4. stack-4: rename generate/stream to generate_async/stream_async (#1769)
5. stack-5: remove LangChain imports from core modules (#1770)
6. **stack-6: move LangChain implementations into integrations/langchain/** (this PR)
7. stack-7: framework-owned provider registry (#1773)
8. stack-8: framework-agnostic test infrastructure (TODO)

## Description

Move providers, helpers, langchain_initializer, and safetools into integrations/langchain/. Keep llm/providers/__init__.py as a backwards compatible shim for public API (register_chat_provider, etc.). Update all test imports and patch paths to new locations.

The shim is temporary. Stack 7 will introduce a framework-agnostic provider registry in core where each framework (LangChain, etc.) registers its own providers. At that point the shim is replaced with the real generic API.

## Summary by CodeRabbit

* **Documentation**
  * Updated streaming configuration examples with corrected import paths.

* **Refactor**
  * Reorganized internal module structure for better code organization. Old import paths now display clear migration guidance.

* **Chores**
  * Updated copyright year information.

### CODE REVIEW

**Comment by Pouyanpi** [comment]
@greptileai

these are pre-existing issues and not related to this PR and thus out of scope:



> P1 — _parse_version crashes on pre-release installs: int() is applied to every dot-split segment; version strings like \"1.0.0rc1\" or \"2.0.0.post1\" raise an uncaught ValueError that propagates out of init_langchain_model, blocking all model initialization for users on pre-release LangChain packages.
> P2 — shared kwargs dict mutated across initializer attempts in _update_model_kwargs; if _init_community_chat_models mutates and then fails, _init_text_completion_model sees the polluted dict.

File: nemoguardrails/integrations/langchain/actions/__init__.py [hunk 1]
```diff
@@ -1,4 +1,4 @@
-# SPDX-FileCopyrightText: Copyright (c) 2023-2026 NVIDIA CORPORATION & AFFILIATES. All rights reserved.
+# SPDX-FileCopyrightText: Copyright (c) 2026 NVIDIA CORPORATION & AFFILIATES. All rights reserved.
```
Comment (tgasser-nv) [inline nemoguardrails/integrations/langchain/actions/__init__.py:1]: nit: How did the copyright text end up changing from 2023 - 2026 to just 2026, I think we need to change it back

File: nemoguardrails/integrations/langchain/providers/huggingface/__init__.py
Comment (tgasser-nv) [inline nemoguardrails/integrations/langchain/providers/huggingface/__init__.py:14]: Somehow the contents of this file were lost in the move/rename:

```python
from .pipeline import HuggingFacePipelineCompatible
from .streamers import AsyncTextIteratorStreamer

__all__ = [
    "HuggingFacePipelineCompatible",
    "AsyncTextIteratorStreamer",
]
```

Without this, imports like this won't work:

```
from nemoguardrails.integrations.langchain.providers.huggingface import HuggingFacePipelineCompatible
```

File: nemoguardrails/llm/providers/__init__.py [hunk 2]
```diff
@@ -13,7 +13,9 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 
-from .providers import (
+# Re-export from new location for backwards compatibility.
```
Comment (tgasser-nv) [inline nemoguardrails/llm/providers/__init__.py:16]: Do you need to do the same shim approach for the other subdirectories?

1. nemoguardrails/llm/helpers.py (get_llm_instance_wrapper)-> nemoguardrails.integrations.langchain.helpers.
2. nemoguardrails/actions/langchain/__init__.py-> nemoguardrails.integrations.langchain.actions
3. nemoguardrails/actions/langchain/actions.py -> nemoguardrails.integrations.langchain.actions.actions
4. nemoguardrails/actions/langchain/safetools.py -> nemoguardrails.integrations.langchain.actions.safetools

File: tests/llm/models/test_langchain_init_scenarios.py [hunk 3]
```diff
@@ -51,13 +51,13 @@
 
 import pytest
 
-from nemoguardrails.llm.models.langchain_initializer import (
+from nemoguardrails.integrations.langchain.langchain_initializer import (
```
Comment (tgasser-nv) [inline tests/llm/models/test_langchain_init_scenarios.py:54]: Can you add a few tests that import from the shim to cover the shim imports? 

`from nemoguardrails.integrations.langchain.providers import` should be `from nemoguardrails.llm.providers import ...` to make sure the shim works

**Review by tgasser-nv — APPROVED** [review]
Looks good! Can you address the feedback before merging? There are issues with the header copyright, missing backwards-compatibility shims for llm.helpers, etc missing shim unit-tests and empty huggingface __init__..py

File: nemoguardrails/integrations/langchain/actions/__init__.py (hunk 1 above)
Comment (Pouyanpi) [reply nemoguardrails/integrations/langchain/actions/__init__.py:1]: for the pre-commit hook this is a new file and it auto updates the header. I'll see if we can avoid it but I don't think so.

**Comment by Pouyanpi** [comment]
Thank you @tgasser-nv for catching these, all valid and great points 🚀 

  fixed:
  - restored __init__.py re-exports for providers/huggingface and providers/trtllm
  - added import error stubs at the old paths (llm/helpers.py, llm/providers/huggingface, llm/providers/trtllm) will raise ImportError with the new path instead of silently breaking. and no LangChain imported into core.
  - added providers shim import test
  - each stub is tagged TODO(0.24.0) for removal next release

File: nemoguardrails/integrations/langchain/actions/__init__.py (hunk 1 above)
Comment (Pouyanpi) [reply nemoguardrails/integrations/langchain/actions/__init__.py:1]: I'll take care of it
